src/autolabel/tasks/classification.py

```diff
@@ -55,6 +60,40 @@ def construct_prompt(self, input: Dict, examples: List) -> str:
         # prepare task guideline
         labels_list = self.config.labels_list()
         num_labels = len(labels_list)
+
+        # if large number of labels, filter labels_list by similarity of labels to input
+        if num_labels >= 50:
+            example_prompt = PromptTemplate(
+                input_variables=["input"],
+                template="{input}",
+            )
+            label_examples = [{"input": label} for label in labels_list]
+
+            example_selector = SemanticSimilarityExampleSelector.from_examples(
+                # This is the list of labels available to select from.
+                label_examples,
+                # This is the embedding class used to produce embeddings which are used to measure semantic similarity.
+                OpenAIEmbeddings(),
+                # This is the VectorStore class that is used to store the embeddings and do a similarity search over.
+                VectorStoreWrapper(cache=False),
+                # This is the number of examples to produce.
+                k=10,
+            )
+            similar_prompt = FewShotPromptTemplate(
+                example_selector=example_selector,
+                example_prompt=example_prompt,
+                prefix="Input: {example}",
```

Comment:
use the example template from the config here? see how the seed examples are prepared

Reply:
This was unnecessary and has been removed in latest commit.